File: Schedule.Core/DTO/Studying/GroupDto.cs
using Schedule.Core.DTO.Account;
using Schedule.Core.Entities.Studying;
using System;
using System.Collections.Generic;

namespace Schedule.Core.DTO.Studying
{
    public class GroupDto
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public Faculty Faculty { get; set; }
        public IEnumerable<UserDto> Users { get; set; }
        public IEnumerable<Assignment> Assignments { get; set; }
        public IEnumerable<Appointment> Appointments { get; set; }
    }
}

File: Schedule/ClientApp/App/src/app/studying/models/GroupDto.ts
import { Guid } from 'guid-typescript/dist/guid';
import { AppointmentDto } from './AppiontmentDto';
import { UserDto } from './UserDto';

export class GroupDto {
  id: Guid;
  name: string;
  users = new Array<UserDto>();
  membersCount: number;
  appointments = new Array<AppointmentDto>();

  constructor(model: Partial<GroupDto>) {
    this.id = model.id;
    this.name = model.name;
    this.users = model.users;
    this.appointments = model.appointments;
    this.membersCount = model.membersCount;
  }
}
